pkgs/applications/emulators/xemu/default.nix

```diff
@@ -0,0 +1,136 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, qemu
+, makeDesktopItem
+, copyDesktopItems
+, pkg-config
+, python3
+, ninja
+, meson
+, which
+, perl
+, wrapGAppsHook
+, glib
+, gtk3
+, libpcap
+, openssl
+, libepoxy
+, libsamplerate
+, SDL2
+, SDL2_image
+, mesa
+, libdrm
+, libGLU
+, gettext
+, vte
+}:
+
+let
+
+
+
+in stdenv.mkDerivation rec {
+  pname = "xemu";
+  version = "0.7.84";
+
+  src = fetchFromGitHub {
+    owner = "xemu-project";
+    repo = "xemu";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    hash = "sha256-pEXjwoQKbMmVNYCnh5nqP7k0acYOAp8SqxYZwPzVwDY=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    python3
+    python3.pkgs.pyyaml
+    ninja
+    which
+    meson
+    perl
+    wrapGAppsHook
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    glib
+    gtk3
+    openssl
+    mesa
+    libepoxy
+    libdrm
+    libpcap
+    libsamplerate
+    SDL2
+    libGLU
+    SDL2_image
+    gettext
+    vte
+  ];
+
+  separateDebugInfo = true;
+
+  dontUseMesonConfigure = true;
+
+  setOutputFlags = false;
+
+  configureFlags = [
+    "--disable-strip"
+    "--meson=meson"
+    "--target-list=i386-softmmu"
+    "--disable-werror"
+  ];
+
+  buildFlags = [ "qemu-system-i386" ];
+
+  desktopItems = [(makeDesktopItem {
+    name = "xemu";
+    desktopName = "xemu";
+    exec = "xemu";
+    icon = "xemu";
+  })] ;
+
+  preConfigure = let
+    branch = "master";
+    commit = "d8fa50e524c22f85ecb2e43108fd6a5501744351";
+  in ''
```

Comment:
Good versioning feature. I wonder if this should be moved somewhere to top...  by pname + version.

Reply:
Originally, it was between `stdenv.mkDerivation` and the input, but the other guy suggested it should be moved here

Reply:
Ideally the upstream should generate this files (at least for tagged releases) instead of forcing downstream packagers to run such hack.

> Good versioning feature. I wonder if this should be moved somewhere to top...  by pname + version.

It is better to send the variables near the scope they are effectively employed.